src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -428,7 +453,62 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
     }
   }
 
-  override def handleCallRestarts(restartableWorkflows: Seq[RestartableWorkflow])(implicit ec: ExecutionContext): Future[Any] = Future("FIXME")
+  /**
+   * <ul>
+   *   <li>Any execution in Failed should fail the restart.</li>
+   *   <li>Any execution in Aborted should fail the restart.</li>
+   *   <li>Scatters in Starting should fail the restart.</li>
+   *   <li>Collectors in Running should be set back to NotStarted.</li>
+   *   <li>Calls in Starting should be rolled back to NotStarted.</li>
+   *   <li>Calls in Running with no job key should be rolled back to NotStarted.</li>
+   * </ul>
+   *
+   * Calls in Running *with* a job key should be left in Running.  The WorkflowActor is responsible for
+   * resuming the CallActors for these calls.
+   */
+  override def prepareForRestart(restartableWorkflow: WorkflowDescriptor)(implicit ec: ExecutionContext): Future[Unit] = {
+
+    lazy val tag = s"Workflow ${restartableWorkflow.id.shortString}:"
+
+    def handleExecutionStatuses(executions: Traversable[Execution]): Future[Unit] = {
+
+      def stringifyExecutions(executions: Traversable[Execution]): String = {
+        executions.toSeq.sortWith((lt, rt) => lt.callFqn < rt.callFqn || (lt.callFqn == rt.callFqn && lt.index < rt.index)).mkString(" ")
+      }
+
+      def isRunningCollector(key: Execution) = key.index.toIndex.isEmpty && key.executionStatus == ExecutionStatus.Running
+
+      val failedOrAbortedExecutions = executions filter { x => x.executionStatus == ExecutionStatus.Aborted || x.executionStatus == ExecutionStatus.Failed }
+
+      if (failedOrAbortedExecutions.nonEmpty) {
+        Future.failed(new Throwable(s"$tag Cannot restart, found Failed and/or Aborted executions: " + stringifyExecutions(failedOrAbortedExecutions)))
+      } else {
+        // Cromwell has execution types: scatter, collector, call.
+        val (scatters, collectorsAndCalls) = executions partition { _.isScatter }
+        val startingScatters = scatters filter { _.executionStatus == ExecutionStatus.Starting }
+        if (startingScatters.nonEmpty) {
+          Future.failed(new Throwable(s"$tag Cannot restart, found scatters in Starting status: " + stringifyExecutions(startingScatters)))
```

Comment:
I know you probably just haven't gotten to it yet but I think here and the case below are both the cases where it's very rare but the code covers it defensively